src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -569,8 +579,36 @@ private void fixColumnCases(Schema recaptured) throws SQLException {
 			this.shouldSnapshotNextSchema = true;
 	}
 
+	private void fixColumnLength(Schema recaptured) throws SQLException {
+		int colLengthDiffs = 0;
+
+		for ( Pair<ColumnDef, ColumnDef> pair : schema.matchColumns(recaptured) ) {
+			ColumnDef cA = pair.getLeft();
+			ColumnDef cB = pair.getRight();
+
+			if (cA instanceof ColumnDefWithLength) {
+				if (cB != null && cB instanceof ColumnDefWithLength) {
+					long aColLength = ((ColumnDefWithLength) cA).getColumnLength();
+					long bColLength = ((ColumnDefWithLength) cB).getColumnLength();
+
+					if ( aColLength != bColLength ) {
+						colLengthDiffs++;
+						LOGGER.info("correcting column length of `" + cA.getName() + "` to `" + cB.getName() + "`.  Will save a full schema snapshot after the new DDL update is processed.");
+						((ColumnDefWithLength) cA).setColumnLength(bColLength);
+					}
+				} else {
+					LOGGER.warn("warning: Couldn't check for column length on column " + cA.getName() +
+						".  You may want to recapture your schema");
+				}
+			}
+
+			if ( colLengthDiffs > 0 )
+				this.shouldSnapshotNextSchema = true;
+		}
+	}
+
 	protected void handleVersionUpgrades(Connection conn) throws SQLException, InvalidSchemaError {
-		if ( this.schemaVersion < 2 ) {
+		if ( this.schemaVersion < 3 ) {
```

Comment:
need to bump the SchemaStoreVersion constant to 3, so that newly created schemas are saved properly.